#Configuration file for starting up 2 Lidar and merging 2 lidar
urg_node2_1st:
  ros__parameters:
    ip_address: '192.168.0.10'
    ip_port: 10940
    frame_id : 'left_lidar'
    calibrate_time : false
    synchronize_time : false
    publish_intensity : false
    publish_multiecho : false
    error_limit : 4
    error_reset_period : 5.0
    diagnostics_tolerance : 0.05
    diagnostics_window_time : 5.0
    time_offset : 0.0
    angle_min : -3.14
    angle_max : 3.14
    skip : 0
    cluster : 1

urg_node2_2nd:
  ros__parameters:
    ip_address: '192.168.0.20'
    ip_port: 10940
    frame_id : 'right_lidar'
    calibrate_time : false
    synchronize_time : false
    publish_intensity : false
    publish_multiecho : false
    error_limit : 4
    error_reset_period : 5.0
    diagnostics_tolerance : 0.05
    diagnostics_window_time : 5.0
    time_offset : 0.0
    angle_min : -3.14
    angle_max : 3.14
    skip : 0
    cluster : 1

ros2_laser_scan_merger:
  ros__parameters:
    flip1: false
    flip2: false
    inverse1: false
    inverse2: false
    laser1Alpha: 90.0
    laser1AngleMax: 80.0
    laser1AngleMin: -115.0
    laser1B: 0
    laser1G: 0
    laser1R: 255
    laser1XOff: 0.5
    laser1YOff: 0.255
    laser1ZOff: 0.28
    laser2Alpha: -90.0
    laser2AngleMax: 115.0
    laser2AngleMin: -80.0
    laser2B: 255
    laser2G: 0
    laser2R: 0
    laser2XOff: 0.5
    laser2YOff: -0.255
    laser2ZOff: 0.28
    pointCloudTopic: cloud_in
    pointCloutFrameId: base_link
    qos_overrides./parameter_events.publisher.depth: 1000
    # qos_overrides./parameter_events.publisher.durability: volatile
    # qos_overrides./parameter_events.publisher.history: keep_last
    qos_overrides./parameter_events.publisher.reliability: best_effort
    scanTopic1: scan_left
    scanTopic2: scan_right
    show1: true
    show2: true
    use_sim_time: false

pointcloud_to_laserscan:
  ros__parameters:
    angle_increment: 0.0087
    # angle_max: 1.5708
    # angle_min: -1.5708
    angle_max: 3.14
    angle_min: -3.14
    inf_epsilon: 1.0
    max_height: 1.0
    min_height: 0.0
    qos_overrides:
      /parameter_events:
        publisher:
          depth: 1000
          # durability: volatile
          # history: keep_last
          reliability: best_effort
    queue_size: 16
    range_max: 10.0
    range_min: 0.35
    scan_time: 0.3333
    target_frame: base_footprint
    transform_tolerance: 0.01
    use_inf: true
    use_sim_time: false
